concent_api/gatekeeper/tests/test_gatekeeper_views.py

```diff
@@ -0,0 +1,141 @@
+            operation                       = 'download'
+        )
+
+    @freeze_time("2017-11-17 10:00:00")
+    def test_download_should_accept_valid_message(self):
+        golem_upload_token = dump(self.upload_token, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY)
+        encrypted_token = b64encode(golem_upload_token).decode()
+        response = self.client.get('/gatekeeper/download-auth/blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
+            HTTP_AUTHORIZATION = 'Golem ' + encrypted_token,
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key
+        )
+
+        self.assertIsInstance(response, HttpResponse)
+        self.assertEqual(response.status_code, 200)
+
+    @freeze_time("2017-11-17 10:00:00")
+    def test_download_should_return_401_if_wrong_authorization_header(self):
+        golem_upload_token = dump(self.upload_token, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY)
+        encrypted_token = b64encode(golem_upload_token).decode()
+        response1 = self.client.get('/gatekeeper/download-auth/blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
+            HTTP_AUTHORIZATION = 'GolemGolem ' + encrypted_token,
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key
+        )
+
+        self.assertIsInstance(response1, JsonResponse)
+        self.assertEqual(response1.status_code, 401)
+        self.assertIn('message', response1.json().keys())
+
+        response2 = self.client.get('/gatekeeper/download-auth/blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
+            HTTP_AUTHORIZATIONN = 'Golem ' + encrypted_token,
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key
+        )
+
+        self.assertIsInstance(response2, JsonResponse)
+        self.assertEqual(response2.status_code, 401)
+        self.assertIn('message', response2.json().keys())
+
+        response3 = self.client.get('/gatekeeper/download-auth/blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
+            HTTP_AUTHORIZATION = 'Golem ' + 'encrypted_token',
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key
+        )
+
+        self.assertIsInstance(response3, JsonResponse)
+        self.assertEqual(response3.status_code, 401)
+        self.assertIn('message', response3.json().keys())
+
+    @freeze_time("2017-11-17 10:00:00")
+    def test_download_should_return_401_if_wrong_token_cluster_address(self):
+        upload_token = self.upload_token
+        upload_token.storage_cluster_address = 'www://storage.concent.golem.network/'
+        golem_upload_token = dump(upload_token, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY)
+        encrypted_token = b64encode(golem_upload_token).decode()
+        response = self.client.get('/gatekeeper/download-auth/blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
```

Comment:
Don't hard-code paths. The `/download-auth/` part should be taken from a call to `reverse()`.

Reply:
Done. Switch to:
```
response = Client().post(
            '{}{}'.format(
                reverse('gatekeeper:upload'),
                'blender/benchmark/test_task/scene-Helicopter-27-cycles.blend'
            ),
```